internal/git_clone_test.go

```diff
@@ -0,0 +1,76 @@
+package internal
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/databricks/cli/libs/git"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAccGitClone(t *testing.T) {
+	t.Log(GetEnvOrSkipTest(t, "CLOUD_ENV"))
+
+	tmpDir := t.TempDir()
+	ctx := context.Background()
+	var err error
+
+	err = git.Clone(ctx, "https://github.com/ShreyasGoenka/empty-databricks-cli-repo.git", "", tmpDir)
```

Comment:
I use an external repo here because I could not find a databricks repo with the following qualities:
1. Stable
2. Multiple branches
3. Public
4. Small in size

I would also like to test with non databricks repos if are supporting `expandUrl` functionality

Reply:
done